alws/routers/uploads.py

```diff
@@ -18,23 +21,39 @@
     dependencies=[Depends(get_current_user)],
 )
 
+async def get_repo_object(session: AsyncSession, repository):
+    products = (await session.execute(
+        select(Product).filter(
+            Product.repositories.any(Repository.name == repository)
+            )
+        )).scalars().all()
+    if not products:
+        products = []
+        builds = (await session.execute(
+            select(Build)
+            .filter(Build.repos.any(Repository.name == repository))
+            .options(selectinload(Build.products))
+            )).scalars().all()
+        for build in builds:
+            products.append(build.products)
+    return products
 
 @router.post("/upload_repometada/")
 async def upload_repometada(
     modules: typing.Optional[UploadFile] = None,
     comps: typing.Optional[UploadFile] = None,
     repository: str = Form(...),
-    product_name: str = Form(...),
     session: AsyncSession = Depends(get_db),
     user: User = Depends(get_current_user),
 ):
     msg = ""
     uploader = MetadataUploader(session, repository)
-    product = get_products(session, product_name=product_name)
-    if not can_perform(product, user, actions.ReleaseToProduct.name):
-        raise PermissionDenied(
-            "User does not have permissions to upload repository metadata."
-        )
+    repo_objects = await get_repo_object(session, repository)
```

Comment:
Looking at uploader code, it currently looks for the repository in pulp directly, see https://github.com/AlmaLinux/albs-web-server/blob/4549f790ae4f8a56ee69417562dd95c9cb74e28a/alws/utils/uploader.py#L276-L278

For this reason, we'd need to do some additional calculation here to figure out the repository in ALBS db, with the provided pulp repo name.

In pulp, production repos naming convention is `distro_name-distro_major_ver-stream-arch`, i.e.: `almalinux-8-baseos-x86_64`.

Naming convention for product repos is: `product_name-distro_name-distro_major_ver-arch-[debug]-dr`, i.e.:
```
soksanichenko-test_product-almalinux-9-x86_64-dr
soksanichenko-test_product-almalinux-9-aarch64-dr
soksanichenko-test_product_2-almalinux-8-aarch64-dr
soksanichenko-test_product_2-almalinux-9-i686-debug-dr
soksanichenko-test_product_2-almalinux-9-i686-dr
soksanichenko-test_product_2-almalinux-9-x86_64-debug-dr
```

Naming convention for build repos is: `distro_name-distro_major_ver-arch-build_id-[debug]-br`, i.e.:
```
AlmaLinux-8-x86_64-76-debug-br
AlmaLinux-9-ppc64le-27-br
AlmaLinux-9-x86_64-28-br
AlmaLinux-8-src-28-br
AlmaLinux-9-x86_64-28-debug-br
```

So, we'd need to first guess the repo (in ALBS) we are going to deal with and then figure out the product it belongs to.